### DISCUSSION TITLE
Detect write-only attributes (mutators)

### DISCUSSION THREAD

**Discussion by pikant** [spatie/laravel-model-info#12] [General]
While tinkering around with this package, I noticed that it detects [Laravel 9 style accessors and mutators](https://laravel.com/docs/9.x/eloquent-mutators#accessors-and-mutators) as well as [Laravel 8 and below style accessors](https://laravel.com/docs/8.x/eloquent-mutators#defining-an-accessor).

However, it does not detect [Laravel 8 and below style **mutators**](https://laravel.com/docs/8.x/eloquent-mutators#defining-a-mutator). In other words, something like this snippet here will be ignored when there is no matching accessor:

```php
    public function setFirstNameAttribute($value)
    {
        $this->attributes['first_name'] = strtolower($value);
    }
```

Is there a particular reason for this behavior or was this simply not implemented yet?

I ask, because I currently try to write a very simple alternative to [barryvdh/laravel-ide-helper](https://github.com/barryvdh/laravel-ide-helper) that works better for multi-tenancy apps.

**Comment by freekmurze** [comment]
It's simply not implemented yet. Feel free to PR support, make sure to include tests.

**Reply by pikant** [reply]
Thanks for the quick answer. I will create a PR.

Quick question on the `cast_type`. For me, it would make sense to have the following cases:

1) `cast_type` => `attribute`
If the model uses Laravel 9 style attributes **or** an accessor, as well as a mutator method, was found.

2) `cast_type` => `accessor`
Just an accessor method was found.

3) `cast_type` => `mutator` ***new**
Just a mutator method was found.

Any complains?
